components/inputs/input-date-range.js

```diff
@@ -129,16 +149,71 @@ class InputDateRange extends RtlMixin(LocalizeCoreElement(LitElement)) {
 		if (input) input.focus();
 	}
 
+	async validate(showErrors) {
+		const errors = await super.validate(showErrors);
+		if (errors.length !== 0) {
+			return errors;
+		}
+		setTimeout(async() => {
+			return Promise.all([
+				await this.shadowRoot.querySelector('.d2l-input-date-range-start').validate(showErrors),
+				await this.shadowRoot.querySelector('.d2l-input-date-range-end').validate(showErrors)]
+			).then((res) => {
+				return res.reduce((acc, errors) => [...acc, ...errors], []);
+			});
+		});
+	}
+
+	get validationMessageBadInput() {
+		return this.localize('components.input-date-range.errorBadInput', { startLabel: this._startLabel, endLabel: this._endLabel });
+	}
+
+	get _endLabel() {
+		return this.endLabel ? this.endLabel : this.localize('components.input-date-range.endDate');
+	}
+
 	async _handleChange(e) {
-		// TODO: validation that start is before end
 		const elem = e.target;
-		if (elem.classList.contains('d2l-input-date-range-start')) this.startValue = elem.value;
-		else this.endValue = elem.value;
+		if (elem.classList.contains('d2l-input-date-range-start')) {
+			this.startValue = elem.value;
+		} else {
+			this.endValue = elem.value;
+		}
+		await this.requestValidate();
 		this.dispatchEvent(new CustomEvent(
 			'change',
 			{ bubbles: true, composed: false }
 		));
 	}
 
+	_handleDropdownOpen(e) {
+		const id = e.target.id;
+		const tooltip = this.shadowRoot.querySelector(`d2l-tooltip[for="${id}"]`);
+		if (tooltip && tooltip.showing) tooltip.hide();
+	}
```

Comment:
We might be able to replace this with tooltip `close-on-click`. It was introduced so that when a button gets interacted with to open a dropdown the tooltip will close.

Reply:
So this kind of works, but it does only close on click and not on enter or down arrow.

Reply:
Thats unfortunate, it would be really nice to be able to handle it without an event. We could add something like `close-on-keypress=[key-code-1, ..., key-code-n]` but not sure whether that's something we'd want in tooltip.